### PR TITLE
Broker documentation updates for android

### PR DESCRIPTION
**Pull request by trwalke**
Adding more info on intent filters and the redirect URI

### CODE REVIEW

File: 2-With-broker/README.md [hunk 1]
```diff
@@ -327,21 +329,31 @@ If you are using the system browser for interactive authntication, is possible t
...
 //This uses the base64 encoded signature produced above.
-<intent-filter>
-      <data android:scheme="msauth"
-                    android:host="Package Name"
+      <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="msal"Client ID"" android:host="auth" />
+        <data android:scheme="msauth"
```
Comment (trwalke) [inline 2-With-broker/README.md:337]: we already ask customers to add an intent filter to handle the system browser . however, when broker is used the redirect URI is different so we ended up asking the customer to add a second intent filter to handle the broker redirect URI as well. the issue here is that the intent filters can ovveride each other based on the order and break the functionality entirely. the solution is to combine the intent filters as shown here.

**Comment by trwalke** [comment]
@aiwangmicrosoft will discuss these in our sync

File: 2-With-broker/README.md [hunk 2]
```diff
@@ -315,6 +315,8 @@ More information on the [keytool here](https://docs.oracle.com/cd/E82085_01/140/
 
 Once you have your signature, simply use the `msauth://"Package Name"/"Signature Hash"` format as shown above to create your redirect URI.
 
+NOTE: if you are still having trouble calculating the correct redirect URI for the broker, MSAL will thrown an exception with the correct broker redirect URI in the error message. if you see this exception, simply update your redirect URI in the code and portal with the one acquired fromt the exception as shown above.
```
Comment (trwalke) [inline 2-With-broker/README.md:318]: add line where exception is caught

File: 2-With-broker/README.md (hunk 1 above)
Comment (bgavrilMS) [reply 2-With-broker/README.md:337]: I think these changes should not only be documented, but be made part of the sample.

Comment (jmprieur) [reply 2-With-broker/README.md:337]: I agree!

Comment (trwalke) [reply 2-With-broker/README.md:337]: The issue here is that the changes require a specific signature from the broker redirect URI that is generated at build time. Unfortunately we do not know what that will be so I would prefer not to add this entry to the sample with a blank signature. I wish there was a better way of handling it but I'm not sure what we can do here. Any other suggestions to improve this?

@jmprieur @bgavrilMS

Comment (bgavrilMS) [reply 2-With-broker/README.md:337]: Let's configure the embedded browser fallback and add a link to the docs for people wanting to use the system browser fallback?

**Pull request merged**